File: Tarea7/src/javaapplication5/Employee.java
package javaapplication5;
import java.util.ArrayList;
public class Employee {

   private String name;
    private int age;
    private double salary;
    private ArrayList<BankAccount> accounts;

    public Employee(String name, int age, double salary) {
        this.name = name;
        this.age = age;
        this.salary = salary;
        this.accounts = new ArrayList<>();
    }

    public Employee(String name, int age, double salary, long accountNumber) {
        this.name = name;
        this.age = age;
        this.salary = salary;
        this.accounts = new ArrayList<>();
        this.accounts.add(new BankAccount(accountNumber));
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        if (age >= 18 && age <= 65) {
            this.age = age;
        } else {
            System.out.println("Edad Invalida");
        }
    }

    public double getSalary() {
        return salary;
    }

    public void setSalary(double salary) {
        if (salary > 0) {
            this.salary = salary;
        } else {
            System.out.println("Salario Invalido");
        }
    }

    public ArrayList<BankAccount> getAccounts() {
        return accounts;
    }

    public String getData() {
        return String.format("Nombre: %s, Edad: %d, Salario: %.2f", this.getName(), this.getAge(), this.getSalary());
    }
}

File: Tarea7python/Employee.py
class BankAccount:
    def __init__(self, account_number):
        self.account_number = account_number


class Employee:
    def __init__(self, name, age, salary):
        self.name = name
        self.age = age
        self.salary = salary
        self.accounts = []

    def get_name(self):
        return self.name

    def get_age(self):
        return self.age

    def get_salary(self):
        return self.salary

    def get_accounts(self):
        return self.accounts

    def get_data(self):
        return f"Nombre: {self.get_name()}, Edad: {self.get_age()}, Salario: {self.get_salary()}"

    def add_account(self, account):
        self.accounts.append(account)


# Ejemplo de uso:
# employee1 = Employee("John Doe", 30, 50000)
# employee2 = Employee("Jane Smith", 25, 60000, 123456789)
# print(employee1.get_data())
# print(employee2.get_data())
